libtransmission/torrent.cc

```diff
@@ -2180,7 +2180,7 @@ static void deleteLocalData(tr_torrent* tor, tr_fileFunc func)
     ****  Move the local data to a new tmpdir
     ***/
 
-    auto tmpdir = tr_strvPath(top, TR_PATH_DELIMITER_STR, tr_torrentName(tor), "__XXXXXX");
+    auto tmpdir = tr_strvPath(top, tr_torrentName(tor) + "__XXXXXX"s);
```

Comment:
Another way would be
```suggestion
    auto tmpdir = tr_strvJoin(top, TR_PATH_DELIMITER_STR, tr_torrentName(tor), "__XXXXXX"sv);
```
and I see it in a couple of places, but not sure if it's the right thing to do given that we were almost ready to use `std::filesystem::path` and this approach wouldn't have worked there anyway.